apps/server/src/modules/group/uc/group.uc.ts

```diff
@@ -193,24 +197,49 @@ export class GroupUc {
 	private async mapClassInfosFromClasses(
 		filteredClassesForSchoolYear: { clazz: Class; schoolYear?: SchoolYearEntity }[]
 	): Promise<ClassInfoDto[]> {
-		const classInfosFromClasses = await Promise.all(
+		const classInfosFromClasses: ClassInfoDto[] = await Promise.all(
 			filteredClassesForSchoolYear.map(async (classWithSchoolYear): Promise<ClassInfoDto> => {
-				const teachers: UserDO[] = await Promise.all(
-					classWithSchoolYear.clazz.teacherIds.map((teacherId: EntityId) => this.userService.findById(teacherId))
-				);
+				const { teacherIds } = classWithSchoolYear.clazz;
+				const teachers: UserDO[] = await this.getTeachersByIds(teacherIds, classWithSchoolYear);
+
+				const filteredTeachers: UserDO[] = teachers.filter((teacher: UserDO): teacher is UserDO => teacher !== null);
 
 				const mapped: ClassInfoDto = GroupUcMapper.mapClassToClassInfoDto(
 					classWithSchoolYear.clazz,
-					teachers,
+					filteredTeachers,
 					classWithSchoolYear.schoolYear
 				);
 
 				return mapped;
 			})
 		);
+
 		return classInfosFromClasses;
 	}
 
+	private async getTeachersByIds(
+		teacherIds: EntityId[],
+		classWithSchoolYear: { clazz: Class; schoolYear?: SchoolYearEntity }
```

Comment:
It looks like you only need the classId so you don't have to put in the whole object